api/cds.proto

```diff
@@ -212,6 +212,14 @@ message Cluster {
     // double. That is, if the desired factor is 1.9, the runtime value should
     // be 1900. Defaults to 1900.
     google.protobuf.UInt32Value success_rate_stdev_factor = 9;
+    // The number of consecutive gateway failures (502, 503, 504 status or
```

Comment:
Your intention here is that these include Envoy originated responses also, right? (Basically you are trying to avoid ejecting on 500?)

Reply:
I originally set out to eject only on Envoy connection errors (failures to connect, timeouts, resets, etc), the easiest way to implement that was to eject on 503/504s since Envoy maps connection errors to those status codes. Ejecting on remote responses with those codes was a side effect, but one that feels like it makes sense in this context, as well as simplifying the implementation. I added 502s to the list too since they belong in the same category of errors.

I guess, yes, in short, I'm trying to avoid ejecting on 500s (or other 5xx that are likely generated by the application.)

Reply:
We could add ejection on resets (similar to retries) which would allow a policy for failure to connect, etc. The only issue with this is that it will also include proxied 502/503/504. I don't  have any real objection to this change but just pointing out that if you actually want the other thing we can do that.

Reply:
Thinking it through, I think it makes sense to include proxied 502/503/504, so I'm happy with this change.